File: 숫자문자열과영단어/javascript/20221028/solution.js
const NUMBERS = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];

export function convertToStringNumber(s) {
  const words = s.split('');
  return [...words.keys()].reduce((acc, i) => acc.replaceAll(NUMBERS[i], i), s);
}

export default function solution(s) {
  return Number(convertToStringNumber(s));
}

File: 숫자문자열과영단어/java/20221028/src/Solution.java
import java.util.Arrays;

public class Solution {
    String[] stringNumbers = {"zero", "one", "two",
            "three", "four", "five", "six",
            "seven", "eight", "nine"};

    public String[] index = {
            "0", "1", "2", "3", "4", "5", "6", "7", "8", "9"
    };

    public int solution(String s) {
        String answer = convertToStringNumber(s);

        return Integer.parseInt(answer);
    }

    public String convertToStringNumber(String s) {
        return Arrays.stream(index)
                .reduce(s, (acc, index) -> acc.replace(stringNumbers[Integer.parseInt(index)], index));
    }
}
